src/coreclr/vm/prestub.cpp

```diff
@@ -602,6 +602,24 @@ PCODE MethodDesc::GetPrecompiledR2RCode(PrepareCodeConfig* pConfig)
         }
     }
 #endif
+
+#ifdef FEATURE_MULTICOREJIT
+    if (pCode != NULL && pConfig->NeedsMulticoreJitNotification())
+    {
+        _ASSERTE(pConfig->GetCodeVersion().IsDefaultVersion());
+        _ASSERTE(!pConfig->IsForMulticoreJit());
+
+        MulticoreJitManager & mcJitManager = GetAppDomain()->GetMulticoreJitManager();
+        if (mcJitManager.IsRecorderActive())
+        {
+            if (MulticoreJitManager::IsMethodSupported(this))
+            {
+                mcJitManager.RecordMethodLoad(this); // Tell multi-core JIT manager to record method on successful load from R2R
```

Comment:
In my branch I have moved this code to after a successful `SetNativeCode()` in the caller such that there would not be duplicate recordings of the method during multi-threaded races